### PR TITLE
BIGTOP-1975. Smoke tests for Spark SQL

### PR DESCRIPTION
**Pull request by youngwookim**

### CODE REVIEW

File: bigtop-tests/smoke-tests/spark/TestSpark.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import static org.apache.bigtop.itest.LogErrorsUtils.logError
+
+class TestSpark {
+  static private Log LOG = LogFactory.getLog(Object.class)
+
+  static Shell sh = new Shell("/bin/bash -s")
+  static final String SPARK_HOME = System.getenv("SPARK_HOME")
+  static final String SPARK_SHELL = SPARK_HOME + "/bin/spark-shell --master yarn-client"
+  static final String TEST_SAPRKSQL_LOG = "/tmp/TestSpark_testSparkSQL.log"
```
Comment (rnowling) [inline bigtop-tests/smoke-tests/spark/TestSpark.groovy]: Spelling of variable name

File: bigtop-tests/smoke-tests/spark/TestSpark.groovy [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  @BeforeClass
+  static void setUp() {
+    sh.exec("rm -f " + TEST_SAPRKSQL_LOG)
```
Comment (rnowling) [inline bigtop-tests/smoke-tests/spark/TestSpark.groovy]: Spelling of variable name

File: bigtop-tests/smoke-tests/spark/TestSpark.groovy [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  public static void tearDown() {
+    sh.exec("hdfs dfs -ls")
+    logError(sh)
+    sh.exec("hdfs dfs -rmr people* examples")
+    logError(sh)
+  }
+
+  @Test
+  void testSparkSQL() {
+    sh.exec(SPARK_SHELL + " --class org.apache.spark.examples.sql.JavaSparkSQL " + " --jars " + SPARK_HOME + "/lib/spark-examples*.jar > " + TEST_SAPRKSQL_LOG + " 2>&1")
```
Comment (rnowling) [inline bigtop-tests/smoke-tests/spark/TestSpark.groovy]: Spelling of variable name

**Comment by rnowling** [comment]
Does the JavaSparkSQL example check the correctness of the SQL result?  Or are you just going for a basic "no errors were reported" approach?

**Comment by youngwookim** [comment]
Addressed RJ's comment and updated this PR.

@rnowling Yes, it just checks that the shell returned 0

**Comment by rnowling** [comment]
+1. LGTM.  In the future, maybe we can parse the output log for results but this is a great start.

**Pull request merged**